[2023-05-31T03:31:10.007+0000] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: birthday_email_dag.send_birthday_emails scheduled__2023-05-30T00:00:00+00:00 [queued]>
[2023-05-31T03:31:10.021+0000] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: birthday_email_dag.send_birthday_emails scheduled__2023-05-30T00:00:00+00:00 [queued]>
[2023-05-31T03:31:10.022+0000] {taskinstance.py:1308} INFO - Starting attempt 4 of 4
[2023-05-31T03:31:10.055+0000] {taskinstance.py:1327} INFO - Executing <Task(PythonOperator): send_birthday_emails> on 2023-05-30 00:00:00+00:00
[2023-05-31T03:31:10.060+0000] {standard_task_runner.py:57} INFO - Started process 1825 to run task
[2023-05-31T03:31:10.065+0000] {standard_task_runner.py:84} INFO - Running: ['airflow', 'tasks', 'run', 'birthday_email_dag', 'send_birthday_emails', 'scheduled__2023-05-30T00:00:00+00:00', '--job-id', '234', '--raw', '--subdir', 'DAGS_FOLDER/birthday_dag.py', '--cfg-path', '/tmp/tmpbliedtf1']
[2023-05-31T03:31:10.067+0000] {standard_task_runner.py:85} INFO - Job 234: Subtask send_birthday_emails
[2023-05-31T03:31:10.147+0000] {task_command.py:410} INFO - Running <TaskInstance: birthday_email_dag.send_birthday_emails scheduled__2023-05-30T00:00:00+00:00 [running]> on host ce3f9f28322a
[2023-05-31T03:31:10.298+0000] {taskinstance.py:1547} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='birthday_email_dag' AIRFLOW_CTX_TASK_ID='send_birthday_emails' AIRFLOW_CTX_EXECUTION_DATE='2023-05-30T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='4' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2023-05-30T00:00:00+00:00'
[2023-05-31T03:31:10.300+0000] {sql.py:262} INFO - Executing: SELECT employee_email, employee_name FROM employee WHERE DATE_FORMAT(employee_date_of_birth, '%m-%d') = DATE_FORMAT(NOW(), '%m-%d')
[2023-05-31T03:31:10.312+0000] {base.py:73} INFO - Using connection ID 'hrms' for task execution.
[2023-05-31T03:31:10.540+0000] {base.py:73} INFO - Using connection ID 'hrms' for task execution.
[2023-05-31T03:31:10.545+0000] {sql.py:375} INFO - Running statement: SELECT employee_email, employee_name FROM employee WHERE DATE_FORMAT(employee_date_of_birth, '%m-%d') = DATE_FORMAT(NOW(), '%m-%d'), parameters: None
[2023-05-31T03:31:10.548+0000] {sql.py:384} INFO - Rows affected: 4
[2023-05-31T03:31:10.549+0000] {taskinstance.py:1824} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 181, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 198, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/birthday_dag.py", line 63, in send_birthday_emails
    """.format(image_data=image_data, employee_name=employee_name)
NameError: name 'image_data' is not defined
[2023-05-31T03:31:10.561+0000] {taskinstance.py:1350} INFO - Marking task as FAILED. dag_id=birthday_email_dag, task_id=send_birthday_emails, execution_date=20230530T000000, start_date=20230531T033110, end_date=20230531T033110
[2023-05-31T03:31:10.593+0000] {standard_task_runner.py:109} ERROR - Failed to execute job 234 for task send_birthday_emails (name 'image_data' is not defined; 1825)
[2023-05-31T03:31:10.641+0000] {local_task_job_runner.py:225} INFO - Task exited with return code 1
[2023-05-31T03:31:10.666+0000] {taskinstance.py:2651} INFO - 0 downstream tasks scheduled from follow-on schedule check
